src/work/PokerGame/PokerGame.sol

```diff
@@ -0,0 +1,516 @@
+    uint32 callbackGasLimit = 200000;
+
+    bytes32 public constant PLAYER_ROLE = keccak256("PLAYER_ROLE");
+    bytes32 public constant CURRENT_TURN_ROLE = keccak256("CURRENT_TURN_ROLE");
+    bytes32 public constant DEALER_ROLE = keccak256("DEALER_ROLE");
+
+   // Struct to represent community cards
+    struct CommunityCards {
+        bytes32 flop1Commitment;
+        bytes32 flop2Commitment;
+        bytes32 flop3Commitment;
+        bytes32 turnCommitment;
+        bytes32 riverCommitment;
+        Card flop1;
+        Card flop2;
+        Card flop3;
+        Card turn;
+        Card river;
+        bool flopRevealed;
+        bool turnRevealed;
+        bool riverRevealed;
+    }
+
+    uint8 constant MAX_PLAYERS = 6; // Maximum number of players per game
+    uint256 constant INITIAL_CHIP_COUNT = 1000; // Initial chip count for each player
+
+    // Enum representing the possible states of the game
+    enum GameState { WAITING, SHUFFLING, ACTIVE, REVEALING_COMMUNITY_CARDS, FINISHED }
+
+    // Enum representing the possible actions a player can take
+    enum PlayerAction { NONE, CALL, RAISE, FOLD }
+
+    // Struct representing a player in the game
+    struct Player {
+        address addr; // Player's address
+        uint256 chipCount; // Player's chip count
+        PlayerAction lastAction; // Player's last action
+        bool isActive; // Whether the player is active in the game
+        bool hasPlayerRole;
+        bool hasCurrentTurnRole;
+    }
+
+    // Struct representing a poker game
+    struct Game {
+        uint256 id; // Game ID
+        GameState state; // Current state of the game
+        uint8 playerCount; // Number of players in the game
+        uint8 currentPlayerIndex; // Index of the current player
+        uint256 pot; // Total pot amount
+        uint256 highestBet; // Highest bet in the current round
+        mapping(uint8 => Player) players; // Mapping of players in the game
+        uint8 dealerIndex; // Index of the dealer
+        uint8 smallBlindIndex; // Index of the small blind
+        uint8 bigBlindIndex; // Index of the big blind
+        address dealer;
+        CommunityCards communityCards;
+    }
+
+    mapping(uint256 => Game) public games; // Mapping of game IDs to Game structs
+
+    // Events for logging important actions in the game
+    event GameCreated(uint256 indexed gameId, address creator); // Event for game creation
+    event PlayerJoined(uint256 indexed gameId, address player); // Event for player joining
+    event GameStarted(uint256 indexed gameId); // Event for game starting
+    event TurnChanged(uint256 indexed gameId, address player); // Event for turn changing
+    event PlayerAction(uint256 indexed gameId, address player, PlayerAction action, uint256 amount); // Event for player action
+    event GameEnded(uint256 indexed gameId, address winner, uint256 pot); // Event for game ending
```

Comment:
you can add a method for the players to end the game if the dealer is inactive after a certain block number. This can be a vote.